### PR TITLE
Create active order

### PR DESCRIPTION
**Pull request by steubs**
Creates simple function that takes in params and creates an entry on active_buy_sell_order table. 

Let me know if I'm on the mark here with what we're looking for. Can I expect I'm getting in reasonable input params? Or should all the input be validated here?

### CODE REVIEW

File: backend/routes/api/create_active_order.py [hunk 1]
```diff
@@ -21,4 +21,17 @@ def create_sell_order():
     # Insert a test entry into the active_buy_sell table
     test_entry = test_data.test_entry_2()
     supabase_middleman.insert_entry('active_buy_sell', test_entry)
-    return
\ No newline at end of file
+    return
+
+def create_active_order(user_id: str, buy_or_sell: bool, stock_id, price: float, quantity: int, expirey: str):
```
[the remark this answers is not in the document]
Comment (steubs) [reply backend/routes/api/create_active_order.py:26]: Sounds good, I'll use these types when creating the entry. Couple questions:

1. Is this data getting passed into the function as the types you say (price as int8, buy_or_sell as bool, etc), or are they getting passed as strings and need to be converted to these types?    

2. Is orderId getting passed in or do i generate the uuid for orderId here?

[the remark this answers is not in the document]
Comment (evanFroelich) [reply backend/routes/api/create_active_order.py:26]: we have to generate the uuid for the orderId. and for the first part, that would be a question for the frontend team on how its handled

**Comment by evanFroelich** [comment]
One other thing is that when the order is created that is when we validate that the user can even make the order. 

If the user puts in a buy order then we have to check if they have at least price*quantity amount of currency on their balance and if not, then return an error insufficient funds. If they have the funds then we subtract them and create the order. 

If the user puts in a sell order, we check if they have the quantity of stock to sell. If they have less than the quantity, return an insufficient quantity error. If they have no entry in their portfolio for the stock at all return something like user doesn't own this stock. If they have enough, take the stock away from their portfolio and create the orders.

File: backend/main.py [hunk 2]
```diff
@@ -12,14 +12,39 @@
...
-
-@app.get('/buyOrder')
+@app.get("/buyOrder")
 def test_entry_1():
     create_active_order.create_buy_order()
     return "Test entry inserted"
 
 
-@app.get('/qb')
+@app.get("/qb")
```
Comment (steubs) [inline backend/main.py:21]: from black formatter auto lint

File: backend/routes/api/create_active_order.py [hunk 3]
```diff
@@ -5,33 +5,65 @@
 from datetime import datetime, timedelta
 from util.ext import test_data
 from database.supabase.store import supabase_middleman
-app=FastAPI()
-load_dotenv()
+from uuid import uuid4
 
+app = FastAPI()
```
Comment (steubs) [inline backend/routes/api/create_active_order.py:10]: from black formatter auto lint

**Comment by steubs** [comment]
New commit created after adding black auto formatting, which created changes to not-relevant-to-this-commit code.

1. creates createActiveOrder endpoint 
2. adds some validation of input in createActiveOrder endpoint main.py.  
3. added loop to create N identical entries based on quantity (quant=5 --> 5 entries) 

Yet to address "validate that the user can even make the order" from your last comment

**Pull request closed without merging**